### PR TITLE
docs: add basic-auth guides

### PR DESCRIPTION
**Pull request by aabedraba**
This PR...
adds basic-auth guides
## Changes

-

## Fixes

-

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

**Review by jasmingacic — APPROVED** [review]
Looking good

**Review by povilasv — APPROVED** [review]
Nice!

File: docs/guides/basic-auth.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Authentication
+
+Currently the methods of authentication you can use are any methods that are Header based. That includes Basic authentication, JWT tokens, Bearer tokens, etc.
```
Comment (kylehodgetts) [inline docs/guides/basic-auth.md:3]:
```suggestion
Currently you can use any Header based authentication methods. This includes:
- Basic Authentication
- JWT Tokens
- Bearer Tokens
```

File: docs/guides/basic-auth.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+To use the authentication, you will need to add the following section to your `x-kusk`:
```
Comment (kylehodgetts) [inline docs/guides/basic-auth.md:5]:
```suggestion
To set up authentication, add the following section to your `x-kusk` settings block:
```

File: docs/guides/basic-auth.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    type: basic
+    auth-upstream:
+      host:
+        hostname: basic-auth-svc.default
+        port:8080
+```
+
+The example above authenticates requests to the whole API.
+
+You can also specify different authentication settings for a specific operation or path. The following example shows authentication configuration for a specific operation:
```
Comment (kylehodgetts) [inline docs/guides/basic-auth.md:23]:
```suggestion
You can also specify different authentication settings for specific operations or paths. The following example shows an example authentication configuration for a specific operation:
```

File: docs/guides/basic-auth.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      auth:
+        type: basic
+        auth-upstream:
+          host:
+            hostname: basic-auth-svc.default
+            port:8080
+      ..
+```
+
+As you can see in the example, in the `hostname` section we're referencing a service in our cluster that will handle the authentication of the requests. This flow looks like the following:
```
Comment (kylehodgetts) [inline docs/guides/basic-auth.md:40]:
```suggestion
In the example, in the `hostname` section we're referencing an upstream service in our cluster that will handle request authentication. The flow is as follows: 
```

**Review by kylehodgetts — APPROVED** [review]

**Pull request merged**